### PR TITLE
Natasha/deephyper initial

### PR DESCRIPTION
**Pull request by nkoussa**
This has been tested with PathDSP and GraphDRP. A readme is provided that I think is pretty clear, but let me know if there are suggestions/comments.

### CODE REVIEW

**Comment by adpartin** [comment]
@nkoussa As far as I see, all changes are related to `workflows/deephyper` except for a small change in `config/base`. I think we can go ahead and merge this.

**Comment by nkoussa** [comment]
> @nkoussa As far as I see, all changes are related to `workflows/deephyper` except for a small change in `config/base`. I think we can go ahead and merge this.

Yeah. Can you approve?

**Comment by adpartin** [comment]
> > @nkoussa As far as I see, all changes are related to `workflows/deephyper` except for a small change in `config/base`. I think we can go ahead and merge this.
> 
> Yeah. Can you approve?

@nkoussa Did you see my comment regarding `exist_ok=True` on line 739?

**Comment by nkoussa** [comment]
@adpartin No, there are no unresolved conversations. Am I missing something?

File: workflows/deephyper_hpo/hpo_deephyper_subprocess.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import logging
+import mpi4py
+from deephyper.evaluator import Evaluator, profile
+from deephyper.evaluator.callback import TqdmCallback
+from deephyper.problem import HpProblem
+from deephyper.search.hps import CBO
+from mpi4py import MPI
+import socket
+import hpo_deephyper_params_def
+from improvelib.config.base import Config
```
Comment (jonesse3) [inline workflows/deephyper_hpo/hpo_deephyper_subprocess.py:15]: @nkoussa @adpartin I see that the base.py is no longer there in the config folder.

Comment (nkoussa) [reply workflows/deephyper_hpo/hpo_deephyper_subprocess.py:15]: Thanks, I can see this comment. Will fix this after the PR is approved.

**Comment by adpartin** [comment]
@nkoussa not sure why you don't my message.

If we use `exist_ok=True` in line 739 in `improvelib/config/base` then `if not os.path.isdir(self.output_dir)` is not required. Correct?

File: workflows/deephyper_hpo/hpo_deephyper_subprocess.py (hunk 1 above)
Comment (jonesse3) [reply workflows/deephyper_hpo/hpo_deephyper_subprocess.py:15]: @adpartin Please see here

**Pull request closed without merging**